packages/pros/src/devices/screen.rs

```diff
@@ -0,0 +1,547 @@
+        self.fill(
+            &Text::new(
+                self.writer_buffer.as_str(),
+                TextPosition::Line(self.current_line),
+                TextFormat::Medium,
+            ),
+            Rgb::WHITE,
+        )?;
+
+        self.writer_buffer.clear();
+
+        Ok(())
+    }
+
+    /// Scroll the entire display buffer.
+    ///
+    /// This function effectively y-offsets all pixels drawn to the display buffer by
+    /// a number (`offset`) of pixels.
+    pub fn scroll(&mut self, start: i16, offset: i16) -> Result<(), ScreenError> {
+        bail_on!(PROS_ERR as u32, unsafe {
+            pros_sys::screen_scroll(start, offset)
+        });
+
+        Ok(())
+    }
+
+    /// Scroll a region of the screen.
+    ///
+    /// This will effectively y-offset the display buffer in this area by
+    /// `offset` pixels.
+    pub fn scroll_area(
+        &mut self,
+        x0: i16,
+        y0: i16,
+        x1: i16,
+        y1: i16,
+        offset: i16,
+    ) -> Result<(), ScreenError> {
+        bail_on!(PROS_ERR as u32, unsafe {
+            pros_sys::screen_scroll_area(x0, y0, x1, y1, offset)
+        });
+
+        Ok(())
+    }
+
+    /// Draw a filled object to the screen.
+    pub fn fill(
+        &mut self,
+        shape: &impl Fill<Error = ScreenError>,
+        color: impl IntoRgb,
+    ) -> Result<(), ScreenError> {
+        shape.fill(self, color)
+    }
+
+    /// Draw an outlined object to the screen.
+    pub fn stroke(
+        &mut self,
+        shape: &impl Stroke<Error = ScreenError>,
+        color: impl IntoRgb,
+    ) -> Result<(), ScreenError> {
+        shape.stroke(self, color)
+    }
+
+    /// Wipe the entire display buffer, filling it with a specified color.
+    pub fn erase(color: impl IntoRgb) -> Result<(), ScreenError> {
+        bail_on!(PROS_ERR as u32, unsafe {
+            pros_sys::screen_set_eraser(color.into_rgb().into())
+        });
+        bail_on!(PROS_ERR as u32, unsafe { pros_sys::screen_erase() });
+
+        Ok(())
+    }
+
+    /// Draw a color to a specified pixel position on the screen.
+    pub fn draw_pixel(x: i16, y: i16) -> Result<(), ScreenError> {
+        bail_on!(PROS_ERR as u32, unsafe {
+            pros_sys::screen_draw_pixel(x, y)
+        });
+
+        Ok(())
+    }
+
+    /// Draw a buffer of pixel colors to a specified region of the screen.
+    pub fn draw_buffer<T, I>(
+        &mut self,
+        x0: i16,
+        y0: i16,
+        x1: i16,
+        y1: i16,
+        buf: T,
+        src_stride: i32,
+    ) -> Result<(), ScreenError>
+    where
+        T: IntoIterator<Item = I>,
+        I: IntoRgb,
+    {
+        let raw_buf = buf
+            .into_iter()
+            .map(|i| i.into_rgb().into())
+            .collect::<Vec<_>>();
+
+        bail_on!(PROS_ERR as u32, unsafe {
+            pros_sys::screen_copy_area(x0, y0, x1, y1, raw_buf.as_ptr(), src_stride)
+        });
+
+        Ok(())
+    }
```

Comment:
Unless you add some checks for the buffer of colors being to small, this should be an unsafe function.